### PR TITLE
Add New Insights Card to navigation bar

### LINKED ISSUE #17845: Stats: Move Add New Insights Card to navigation bar
To make managing the Insights view in Stats more prominent, we should remove the "+ Add stats card" row from the bottom of the Insights screen and instead add a settings icon (cog) to the trailing end of the navigation bar.

### PR DESCRIPTION
**Pull request by sla8c**
This PR adds a new gear (cog) icon to the NavBar upper right when on Stats Insights screen, and disappears when tapping through the other tabs like (Days / Weeks / Months etc)

We have decided to leave the existing Add Stats Card row in place for the time being.

Fixes #17845

To test:

### Test 1 - Adding a new stats card through Navigation Bar
1. Go to Stats screen and then click Insights tab
2. Cog (gear) icon should appear in top right
3. Tap the cog (gear) icon and add a new stats card
4. New stats card should appear in Stats Screen / Insights tab

image

### Test 2 - Add new stats card cog icon should be removed when navigating through Days / Weeks / Months / Years
1. Go to Stats screen and then click Insights tab
2. Cog (gear) icon should appear in top right
3. Tap the Days / Weeks / Months / Years tab.  Cog icon should not be visible
4. Tap Insights tab again and Cog icon should be visible

### Test 3 - Add new stats card when no Stats are visible
1. Go to Stats screen and then click Insights tab
2. Remove all stats from the screen
3. Tap Add stats card button
4. New stats card should appear in Stats Screen / Insights tab

## Regression Notes
1. Potential unintended areas of impact
Stats and ability to add a new stats card

2. What I did to test those areas of impact (or what existing automated tests I relied on)
I tested in Simulator and also physical device

3. What automated tests I added (or what prevented me from doing so)
None

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by jkmassel** [comment]
You can test the <strong>Jetpack</strong> changes on this Pull Request by downloading it from AppCenter <a href='https://install.appcenter.ms/orgs/automattic/apps/jetpack-installable-builds/'>here</a> with build number: <code>pr17867-75f970c</code>. IPA is available <a href='https://appcenter-filemanagement-distrib3ede6f06e.azureedge.net/3852a807-c4b9-48da-948f-a3f313281313/Jetpack%20Alpha.ipa?sv=2019-02-02&sr=c&sig=Q3gk4s7GNMjf9jKJsW6kE3gPte7%2B6hA3MjcR8VSHTf8%3D&se=2022-02-05T11%3A37%3A55Z&sp=r'>here</a>. If you need access to this, you can ask a maintainer to add you.

**Comment by jkmassel** [comment]
You can test the <strong>WordPress</strong> changes on this Pull Request by downloading it from AppCenter <a href='https://install.appcenter.ms/orgs/automattic/apps/WPiOS-One-Offs/'>here</a> with build number: <code>pr17867-75f970c</code>. IPA is available <a href='https://appcenter-filemanagement-distrib2ede6f06e.azureedge.net/f144f44c-8611-476a-86b6-4b7aebd2e343/WordPress%20Alpha.ipa?sv=2019-02-02&sr=c&sig=RHzwgFvQWQqcFS58Bm0LRlketzCq3Avh0kVguY6gwO0%3D&se=2022-02-05T11%3A33%3A46Z&sp=r'>here</a>. If you need access to this, you can ask a maintainer to add you.

File: WordPress/Classes/ViewRelated/Stats/SiteStatsDashboardViewController.swift [hunk 1]
```diff
@@ -69,17 +69,41 @@ class SiteStatsDashboardViewController: UIViewController {
...
+        configureNavBar(StatsPeriodType.insights)
         setupFilterBar()
         restoreSelectedDateFromUserDefaults()
         restoreSelectedPeriodFromUserDefaults()
         addWillEnterForegroundObserver()
         view.accessibilityIdentifier = "stats-dashboard"
     }
 
+    func configureNavBar(_ statsPeriodType: StatsPeriodType) {
+        if statsPeriodType == StatsPeriodType.insights {
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Stats/SiteStatsDashboardViewController.swift]: Nitpick, but you could omit `StatsPeriodType` here. We could also make this all into a ternary statement:

```
parent?.navigationItem.rightBarButtonItem = statsPeriodType == .insights ? addStatsButton : nil
```

File: WordPress/Classes/ViewRelated/Stats/SiteStatsDashboardViewController.swift [hunk 2]
```diff
@@ -69,17 +69,41 @@ class SiteStatsDashboardViewController: UIViewController {
     private var periodTableViewController = SiteStatsPeriodTableViewController.loadFromStoryboard()
     private var pageViewController: UIPageViewController?
 
+    @objc lazy var addStatsButton: UIBarButtonItem = {
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Stats/SiteStatsDashboardViewController.swift]: Looking at this now, I wonder if the name could be clearer – `manageInsightsButton` perhaps? If we rename it, we should rename the method it calls to match.

**Review by frosty — COMMENTED** [review]
This works for me! Just left two small suggestions for your consideration.

File: WordPress/Classes/ViewRelated/Stats/SiteStatsDashboardViewController.swift (hunk 2 above)
Comment (sla8c) [reply WordPress/Classes/ViewRelated/Stats/SiteStatsDashboardViewController.swift]: Sounds good!

File: WordPress/Classes/ViewRelated/Stats/SiteStatsDashboardViewController.swift (hunk 1 above)
Comment (sla8c) [reply WordPress/Classes/ViewRelated/Stats/SiteStatsDashboardViewController.swift]: Thanks for your suggestion!  I see I can use the property `currentSelectedPeriod` - thanks + makes sense 😄

File: WordPress/Classes/ViewRelated/Stats/SiteStatsDashboardViewController.swift [hunk 3]
```diff
@@ -69,17 +69,36 @@ class SiteStatsDashboardViewController: UIViewController {
...
         restoreSelectedPeriodFromUserDefaults()
         addWillEnterForegroundObserver()
         view.accessibilityIdentifier = "stats-dashboard"
     }
 
+    func configureNavBar() {
+        parent?.navigationItem.rightBarButtonItem = currentSelectedPeriod == .insights ? manageInsightsButton : nil
+    }
+
+    @objc func addStatsCardButtonTapped() {
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/Stats/SiteStatsDashboardViewController.swift]: Could we rename this method to match the new button name?

**Review by frosty — COMMENTED** [review]
Changes look good, just spotted two more small things! Left one comment in the code, and also when tapping the "Add stats card" row in the table view I'm seeing the tracks event fired twice:

```
Tracked: stats_add_insight_item_tapped <blog_id: 123456789, site_type: blog>
Tracked: stats_add_insight_item_tapped <source: table_row>
```

Can we track down this extra event and remove it? (I guess the first one, as the second event matches the new button)

File: WordPress/Classes/ViewRelated/Stats/SiteStatsDashboardViewController.swift (hunk 3 above)
Comment (sla8c) [reply WordPress/Classes/ViewRelated/Stats/SiteStatsDashboardViewController.swift]: Doh!  Of course done

**Review by frosty — APPROVED** [review]
Looks good!

**Comment by sla8c** [comment]
Updated with keyword `Fixes`

**Pull request merged**